### DISCUSSION TITLE
Explore alternatives to Heroku for streamlit app hosting

### DISCUSSION THREAD

**Discussion by dcaribou** [dcaribou/transfermarkt-datasets#212] [General]
Heroku is discontinuing the free plan by the end of November 2022. The next most economic plan, which seems to be Eco, starts at 5$ per month, which for a low traffic application such as our streamlit app is not perfect.

Explore alternatives to Heroku that can adapt better to the needs of the app.

| Name | Docker support* | Elastic pricing** | AWS Ireland | Custom domains | Test successful
-|-|-|-|-|-
[Heroku Eco](https://devcenter.heroku.com/articles/eco-dyno-hours) | Yes | [No](https://devcenter.heroku.com/articles/eco-dyno-hours) ❌ | Yes | [Yes](https://devcenter.heroku.com/articles/custom-domains)
[Heroku Basic](https://devcenter.heroku.com/articles/eco-dyno-hours) | Yes | [No](https://github.com/dcaribou/transfermarkt-datasets/issues/113#issuecomment-1342554936) ❌ | Yes | [Yes](https://devcenter.heroku.com/articles/custom-domains)
Streamlit Cloud | No ❌ 
[Cyclic](https://www.cyclic.sh/) | No ❌ 
[Deta](https://www.deta.sh/) | No ❌ 
[Fly.io](https://fly.io/) 🏆 | [Yes](https://fly.io/docs/) | [Yes](https://fly.io/docs/about/pricing/) | [No](https://community.fly.io/t/where-is-fly-hosted/1549) | [Yes](https://fly.io/docs/app-guides/custom-domains-with-fly/#teaching-your-app-about-custom-domains) | [Yes](https://transfermarkt-datasets.fly.dev/) ✅
[Render](https://render.com/) | [Yes](https://render.com/docs/docker) | [Yes](https://render.com/docs/free) |  [No](https://render.com/docs/regions) | Yes | [No](https://github.com/dcaribou/transfermarkt-datasets/issues/113#issuecomment-1345375256) ❌
[Railway](https://railway.app/) | [Yes](https://docs.railway.app/deploy/dockerfiles) | [Yes](https://docs.railway.app/reference/plans) | [us-west-1 only](https://discord.com/channels/713503345364697088/1047448563598250025/1047459923186962464) | - | [No](https://github.com/dcaribou/transfermarkt-datasets/issues/113#issuecomment-1345359138) ❌
[Porter](https://porter.run/)
[Clever Cloud](https://www.clever-cloud.com/)
[PythonAnywhere](https://www.pythonanywhere.com/)

*Docker support means we have the ability to create a custom Docker image for the app, and define a launch command **Elastic pricing means either free hosting or a cheap pay as you go option with scale down to zero

**Comment by dcaribou** [comment]
Documentation on heroku website is unclear on whether Heroku Basic
* Is the cost up to 7$ a month, but it it can be less depending on usage?
* Is scale-down to zero available for Basic plan?
If the two above are true, basic/hobby dynos may be a good enough, and we don't need to migrate. It's worth testing it.

**Comment by dcaribou** [comment]
> Is scale-down to zero available for Basic plan?

This is unfortunately not true, hence the basic plan is not an option.

Currently exploring Railway.app

**Comment by dcaribou** [comment]
Railway.app exploration failed: it turns out that Railway.app is based in aws apprunner, on which streamlit [does not seem to work.](https://discuss.streamlit.io/t/new-deployment-option-aws-app-runner/13084)

**Comment by dcaribou** [comment]
Render.com is apprunnner too?

<img width="1284" alt="image" src="https://user-images.githubusercontent.com/16071835/206873536-052c1b69-5b1b-4fcb-9030-29ee9b4bb1b2.png">

**Comment by namtonthat** [comment]
Have you considered using PythonAnywhere? It's owned by Anaconda Inc but they have a commitment to maintaining the Free Tier. 

https://www.pythonanywhere.com/

**Comment by dcaribou** [comment]
> Have you considered using PythonAnywhere? It's owned by Anaconda Inc but they have a commitment to maintaining the Free Tier.
> 
> https://www.pythonanywhere.com/

@namtonthat I hadn't. I'll definitely take a look.
